File: rag_demo/api/FastAPI_implementation/app/schemas/rag.py
from typing import Any

from pydantic import BaseModel, Field


class QueryRequest(BaseModel):
    question: str = Field(..., min_length=1)
    question_type: str | None = None
    include_debug: bool = False


class QueryResponse(BaseModel):
    answer: str
    refusal: bool
    reason: str = ""
    sources: list[int] = []
    gate: dict[str, Any] | None = None
    retrieval_debug: dict[str, Any] | None = None


class IndexResponse(BaseModel):
    chunks: int
    status: str = "completed"


class UploadResponse(BaseModel):
    filename: str
    status: str = "uploaded"

File: rag_demo/web_demo/app/types/rag.ts
export interface RagQueryRequest {
  question: string
  question_type?: string
  include_debug?: boolean
}

export interface RagQueryResponse {
  answer: string
  refusal: boolean
  reason: string
  sources: number[]
  gate: Record<string, unknown> | null
  retrieval_debug: Record<string, unknown> | null
}

export interface RagIndexResponse {
  chunks: number
  status: string
}

export interface RagHealthResponse {
  status: string
}

export interface RagUploadResponse {
  filename: string
  status: string
}

export interface PingResponse {
  status: string
}

export interface ClearResponse {
  status: string
}
